### PR TITLE
New: `prefer-template` rule

### LINKED ISSUE #3014: Rule request: enforce the use of template strings over standard concatenation
It was suggested in #1617 to add a rule to enforce the use of string templates. As far as I know, this has not been implemented yet. 

I was wondering if someone thought about this or began work on this rule?

### PR DESCRIPTION
**Pull request by mysticatea**
Fixes #3014.

### CODE REVIEW

File: lib/rules/prefer-template.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            // Checks whether or not this node had been checked already.
+            if (done[topBinaryExpr.range[0]]) {
+                return;
+            }
+            done[topBinaryExpr.range[0]] = true;
+
+            if (hasNonLiteral(topBinaryExpr)) {
+                context.report(
+                    topBinaryExpr,
+                    "use the template literal instead of strings concatenation.");
```
Comment (xjamundx) [inline lib/rules/prefer-template.js]: I think the typical way of reporting is maybe using something like `Unexpected string concatenation.`

**Comment by IanVS** [comment]
Minor grammar quibble, I think it should be "string concatenation" not "strings concatenation" (singular).

**Comment by mysticatea** [comment]
Thank you for the review!

I modified.

**Comment by nzakas** [comment]
Lgtm

**Pull request merged**